### DISCUSSION TITLE
请问如何裁掉视频的黑边？

### DISCUSSION THREAD

**Discussion by tyf2018** [hooke007/mpv_PlayKit#458] [mpv 通用]
有时候视频会有较大的黑边，其它播放器例如PotPlayer可以拉伸帧位以达到去除黑边的效果，请问
MPV_lazy的配置包有类似的功能吗？找了一下，没找到。之前试过别的lua插件，效果不好。

![image](https://github.com/hooke007/MPV_lazy/assets/56342788/574d6047-5c1f-4236-a088-a2204a047379)

**Comment by hooke007** [comment]
https://github.com/hooke007/MPV_lazy/blob/3bc372885bee5e402f8b27280a477b419f23c5b4/portable_config/input_easy.conf#L56

**Comment by tyf2018** [comment]
@hooke007  谢谢解答。不过很多黑边，cycle-values panscan是无效的。只有拉伸帧位才行。在[input_easy.conf](https://github.com/hooke007/MPV_lazy/blob/3bc372885bee5e402f8b27280a477b419f23c5b4/portable_config/input_easy.conf#L56)中，add video-pan-y 能在垂直方向上移动视频，但我想要在垂直方向上拉伸（add video-zoom能达到类似效果，但会丢失水平方向上的部分内容）。

我想要的效果是像PotPlayer的这个功能一样，能再指点一下吗？
![image](https://github.com/hooke007/MPV_lazy/assets/56342788/f1cb35dd-b0e3-45d9-80cf-fee846d4d8b8)

**Reply by hooke007** [reply]
我不使用pot

**Reply by verygoodlee** [reply]
不太确定你想要的是不是这个效果
> 注意必须在keepaspect=yes下才能生效
```
# 垂直方向拉伸，对应pot里的 帧高+ 帧高-
KEY1 add video-scale-y  0.05
KEY2 add video-scale-y -0.05
# 水平方向拉伸，对应pot里的 帧宽+ 帧宽-
KEY3 add video-scale-x  0.05
KEY4 add video-scale-x -0.05
```
我感觉这种拉伸看起来很难受啊，变胖变瘦，不如用裁切把上下黑边裁掉

**Comment by tyf2018** [comment]
```
> # 垂直方向拉伸，对应pot里的 帧高+ 帧高-
> KEY1 add video-scale-y  0.05
> KEY2 add video-scale-y -0.05
> # 水平方向拉伸，对应pot里的 帧宽+ 帧宽-
> KEY3 add video-scale-x  0.05
> KEY4 add video-scale-x -0.05
```

是的，就是这个！感谢。黑边太多的视频，适当拉伸帧高，效果还是可以的
